application/ui/src/app/pages/matching-group/matching-group.page.ts

```diff
@@ -78,35 +88,107 @@ export class MatchingGroupPage implements OnInit {
       this.presentToast("Please select more than one user!");
     }
     this.clear();
-  }
+    }
   
-  userChangeGroup1(event: {
-    component: IonicSelectableComponent,
-    value: any
-  } ) {
-    console.log(event);
+  onOpenGroup1 ( event: { component: IonicSelectableComponent } ) {
+    let istoogle = this.result_selected_group1.length == 0 ? false : true;
+    this.selectComponentGroup1.toggleItems(istoogle,this.result_selected_group1);
+  }
+
+  onOpenGroup2 ( event: { component: IonicSelectableComponent } ) {
+    let istoogle = this.result_selected_group2.length == 0 ? false : true;
+    this.selectComponentGroup2.toggleItems(istoogle,this.result_selected_group2);
+  }
+
+  addUsersGroup1 () {
```

Comment:
this function is almost the same for group1 and group2
perhaps you can create a common function that takes the group number as parameter and do all?
so that you call the common function from those 2 functions?